admin/react/includes/class-bsf-custom-fonts-admin-ajax.php

```diff
@@ -0,0 +1,297 @@
+			$response_data = array( 'message' => $font_post_id->get_error_message() );
+			wp_send_json_error( $response_data );
+		}
+
+		update_post_meta( $font_post_id, 'fonts-data', $font_data );
+		update_post_meta( $font_post_id, 'fonts-face', $font_face );
+		update_post_meta( $font_post_id, 'font-type', $font_type );
+
+		/**
+		 * Send the response.
+		 */
+		$response_data = array(
+			'message' => __( 'Successfully created the Font!', 'custom-fonts' ),
+		);
+		wp_send_json_success( $response_data );
+	}
+
+	/**
+	 * Edit the existing font post.
+	 *
+	 * @since 2.0.0
+	 */
+	public function bcf_edit_font() {
+
+		$response_data = array( 'message' => $this->get_error_msg( 'permission' ) );
+
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			wp_send_json_error( $response_data );
+		}
+
+		/**
+		 * Nonce verification
+		 */
+		if ( ! check_ajax_referer( 'edit_font_nonce', 'security', false ) ) {
+			$response_data = array( 'message' => $this->get_error_msg( 'nonce' ) );
+			wp_send_json_error( $response_data );
+		}
+
+		$font_data       = isset( $_POST['font_data'] ) ? bcf_sanitize_text_field_recursive( json_decode( stripslashes( $_POST['font_data'] ), true ) ) : array();
+		$font_variations = ! empty( $font_data['variations'] ) ? $font_data['variations'] : array();
+		$font_type       = ! empty( $_POST['font_type'] ) ? sanitize_text_field( $_POST['font_type'] ) : 'local';
+		$font_id         = ! empty( $_POST['font_id'] ) ? absint( $_POST['font_id'] ) : false;
+
+		if ( empty( $font_variations ) || false === $font_id ) {
+			$response_data = array( 'message' => $this->get_error_msg( 'invalid' ) );
+			wp_send_json_error( $response_data );
+		}
+
+		$font_face = bcf_google_fonts_compatibility()->process_google_fonts_locally( $font_data );
+
+		update_post_meta( $font_id, 'fonts-data', $font_data );
+		update_post_meta( $font_id, 'fonts-face', $font_face );
+		update_post_meta( $font_id, 'font-type', $font_type );
+
+		/**
+		 * Send the response.
+		 */
+		$response_data = array(
+			'message' => __( 'Successfully updated the Font!', 'custom-fonts' ),
+		);
+		wp_send_json_success( $response_data );
+	}
+
+	/**
+	 * Delete the font post.
+	 *
+	 * @since 2.0.0
+	 */
+	public function bcf_delete_font() {
+		$response_data = array( 'message' => $this->get_error_msg( 'permission' ) );
+
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			wp_send_json_error( $response_data );
+		}
+
+		/**
+		 * Nonce verification
+		 */
+		if ( ! check_ajax_referer( 'delete_font_nonce', 'security', false ) ) {
+			$response_data = array( 'message' => $this->get_error_msg( 'nonce' ) );
+			wp_send_json_error( $response_data );
+		}
+
+		$font_id = isset( $_POST['font_id'] ) ? absint( $_POST['font_id'] ) : 0;
+
+		if ( ! $font_id ) {
+			$response_data = array( 'message' => $this->get_error_msg( 'invalid' ) );
+			wp_send_json_error( $response_data );
+		}
+
+		$delete_post_response = wp_delete_post( $font_id );
```

Comment:
Before deleting the post, add a check if the post ID is of post-type font.

Reply:
DONE